### DISCUSSION TITLE
Can someone help me with a faulty floor plan?

### DISCUSSION THREAD

**Discussion by hilarytsai** [mozilla/hubs#2870] [General Questions & Discussion] [answered]
I've included a remixable scene here. If possible, I'd love some help on this floor plan. I've tweaked it endlessly and am still having issues. I used to get stuck, so I increased the step height. But now my avatar feels like it's bounding up and down and is way too high.  Could I get someone to take a look and see what might be wrong?  I have no idea if there's a problem with my geometry of if there is some floor plan setting that could work. I've tried lots of combinations of settings. Thank you!

(https://hubs.mozilla.com/scenes/3X6fHch)

**Comment by hilarytsai** [comment, marked as answer]
I think I got it working... odd thing is, that when I hit regenerate, it does not show blue under all walkable areas--yet when I publish, I can move around fine.  Just to be clear, it's the blue mesh that shows walkable surfaces, right? And if there's a gap,  that's where you'll have problems?

**Reply by rawnsley** [reply]
Blue is walkable, but it is sometimes behind the actual scenery. You can save the model from Spoke as a GLTF and look in more detail in an online viewer or Blender where you show and hide the other elements.

**Reply by hilarytsai** [reply]
thanks, rawnsley!  I'll try that.

**Reply by robinkwilson** [reply]
Thanks @rawnsley @hilarytsai for answering! 
I'm marking as answered to get it out of our todo pipeline, please @ me if this needs to be kept open.

**Comment by cvanmeurs** [comment]
I recommend making a custom nav mesh in Blender.  By that I mean, duplicate the floor geometry of your scene, cut out any parts that you don't want people to be able to walk through, bring it into your Spoke scene as a GLB,  and then turn "walkable" off for every item in your Spoke scene EXCEPT for the custom nav mesh geometry, and then generate your floorplan with that.  When the floorplan is generated, turn off the nav mesh item layer prior to publishing.  I'm able to get much more precise results doing this.

**Reply by hilarytsai** [reply]
This is a really great idea, @cvanmeurs!  Thanks for this. I will try this next time! (Should work with Cinema 4D as well, I'm presuming)

**Reply by cvanmeurs** [reply]
Happy to help!  Let me know how it goes! :)
